src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts

```diff
@@ -0,0 +1,323 @@
+        private complaintService: ComplaintService,
+        private translateService: TranslateService,
+        private route: ActivatedRoute,
+        private jhiAlertService: AlertService,
+    ) {
+        translateService.get('artemisApp.assessment.messages.confirmCancel').subscribe((text) => (this.cancelConfirmationText = text));
+    }
+
+    /**
+     * On init set up the route param subscription.
+     * Will load the participation according to participation Id with the latest result and result details.
+     */
+    ngOnInit(): void {
+        // Used to check if the assessor is the current user
+        this.accountService.identity().then((user) => {
+            this.userId = user!.id!;
+        });
+        this.isAtLeastInstructor = this.accountService.hasAnyAuthorityDirect(['ROLE_ADMIN', 'ROLE_INSTRUCTOR']);
+        this.paramSub = this.route.params.subscribe((params) => {
+            this.loadingParticipation = true;
+            this.participationCouldNotBeFetched = false;
+            const participationId = Number(params['participationId']);
+            this.programmingExerciseParticipationService.getStudentParticipationWithResults(participationId).subscribe(
+                (participationWithResults: ProgrammingExerciseStudentParticipation) => {
+                    // Set domain to make file editor work properly
+                    this.domainService.setDomain([DomainType.PARTICIPATION, participationWithResults]);
+                    this.participation = <ProgrammingExerciseStudentParticipation>participationWithResults;
+                    this.automaticResult = this.getLatestAutomaticResult(this.participation.results);
+                    this.manualResult = this.getLatestManualResult(this.participation.results);
+
+                    // Add participation with manual results to display manual result in navbar
+                    this.participationForManualResult = cloneDeep(this.participation);
+                    this.participationForManualResult.results = this.manualResult ? [this.manualResult] : [];
+
+                    // Either latest manual or automatic result
+                    this.submission = this.getLatestResult(this.participation.results)?.submission as ProgrammingSubmission;
+                    this.exercise = this.participation.exercise as ProgrammingExercise;
+
+                    this.checkPermissions();
+
+                    if (this.manualResult && this.manualResult.hasComplaint) {
+                        this.getComplaint();
+                    }
+                },
+                () => {
+                    this.participationCouldNotBeFetched = true;
+                    this.loadingParticipation = false;
+                },
+                () => {
+                    this.loadingParticipation = false;
+                },
+            );
+        });
+    }
+
+    private getLatestResult(results: Result[]): Result | null {
```

Comment:
Just a suggestion: first public, then private methods :-)

Reply:
changed the order of methods :)